### PR TITLE
coord: seal all persisted tables in a single operation for efficiency

### PR DESCRIPTION
**Pull request by danhhz**
The multi-seal ability is newly exposed by persist in the preceding
commit. It would be unacceptably performance to recompute a new
MultiStreamHandle for every call to `advance_local_inputs` so instead
store it on Catalog and recompute it when it changes (tables are added
removed). Then, since we now have a nicely maintained MultiStreamHandle
available, use it in end transaction instead of constructing a one-off
one there.

### CODE REVIEW

File: src/persist/src/indexed/runtime.rs [hunk 1]
```diff
@@ -619,40 +664,40 @@ mod tests {
...
             (c1s1.stream_id(), data[..1].to_vec()),
             (c1s2.stream_id(), data[1..].to_vec()),
         ];
-        block_on(|res| atomic.write_atomic(updates, res))?;
+        block_on(|res| multi.write_atomic(updates, res))?;
         assert_eq!(c1s1_read.snapshot()?.read_to_end(), data[..1].to_vec());
         assert_eq!(c1s2_read.snapshot()?.read_to_end(), data[1..].to_vec());
 
         // Cannot write to streams not specified during construction.
         let (c1s3, _) = client1.create_or_load("3")?;
```
Comment (ruchirK) [inline src/persist/src/indexed/runtime.rs:698]: might be worth adding a seal test here as well?

**Review by ruchirK — APPROVED** [review]
LGTM!

**Comment by danhhz** [comment]
Anecdotally, I ran this locally with mz_metrics set to persisting and sealing seems to be sufficiently fast to keep it from hosing the process (which is not true before this PR).

File: src/persist/src/indexed/runtime.rs (hunk 1 above)
Comment (danhhz) [reply src/persist/src/indexed/runtime.rs:708]: Done

**Review by danhhz — COMMENTED** [review]
Came up with a much more obvious way of keeping the cache invariant, which I've pushed. I feel a lot better about the maintainability of this now

**Review by mjibson — APPROVED** [review]

**Comment by danhhz** [comment]
TFTRs!

**Pull request merged**